src/components/GoodsList/GoodsList.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+
+type Props = {
+  goods: string[],
+};
+
+const GoodsList: React.FC<Props> = ({ goods }) => {
+  const [length, setLength] = useState(1);
+  const [isReverse, setIsReverse] = useState(false);
+  const [sortBy, setSortBy] = useState<string | null>(null);
+  const newGoods = [...goods].filter(good => good.length >= length);
+
+  const reverse = () => {
+    setIsReverse(!isReverse);
+  };
+
+  const sortByName = () => {
+    setSortBy('name');
+    setIsReverse(false);
+  };
+
+  const sortByLength = () => {
+    setSortBy('length');
+    setIsReverse(false);
+  };
+
+  const reset = () => {
+    setLength(1);
+    setSortBy(null);
+    setIsReverse(false);
+  };
+
+  if (sortBy) {
+    newGoods.sort((goodOne, goodTwo) => {
+      switch (sortBy) {
+        case ('name'):
+          return goodOne.localeCompare(goodTwo);
+        case ('length'):
+          return goodOne.length - goodTwo.length;
+        default:
+          return 0;
+      }
+    });
+  }
```

Comment:
Maybe should use useEffect?)

Reply:
I tried, but i can't undarstand how to use 'useEffect' in this situation